app/src/App.js

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { Redirect, Route, Switch } from 'react-router-dom';
+import Loadable from 'react-loadable';
+import TopBarProgress from 'react-topbar-progress-indicator';
+import Layout from 'components/layout';
+
+/* webpackChunkName: "SearchEngine" */
+const SearchEngine = Loadable({
```

Comment:
Is this really necessary? I would guess that importing your own search engine component would be exactly the same as all other components

Reply:
This is known as Code [splitting/chunking](https://medium.com/fabhotels/code-splitting-and-chunking-frontend-apps-with-webpack-a15622388df) based on components to serve them on-demand. Although the current size of the application and files to be downloaded does not require code splitting, it is a technique to improve load time performance since we're only downloading the essential components that the user needs at any given section instead of downloading a monolithic bundle at once.